ram.tf

```diff
@@ -1,3 +1,15 @@
+locals {
+  ram_principals_provided = var.ram_principal != null || length(var.ram_principals) > 0
+  ram_principals = var.ram_resource_share_enabled ? toset(
```

Comment:
I've converted it to a set of strings. Is there any reason you'd prefer a map?

Reply:
map in `for_each` is better b/c in a set, if you update the set, TF will want to recreate all items from the set (this could be OK in many cases, but not acceptable in others, e.g. recreating subnets will not work b/c of all other dependencies)
@alexjurkiewicz

Reply:
I don't think that's accurate. Here's a little test configuration:

```terraform
resource "null_resource" "this" {
  for_each = toset(["a", "b", "c"])
}
```

When I run this, it will create three resources. If you then remove any set item (eg `"b"`) and re-run, only the matching resource will be deleted:

```
  # null_resource.this["b"] will be destroyed
  # (because key ["b"] is not in for_each map)
  - resource "null_resource" "this" {
      - id = "531909274073321123" -> null
    }
```

It won't recreate all items in the set / touch other resources in the for_each.